contracts/Question.sol

```diff
@@ -49,20 +50,28 @@ contract Question {
     donorCount = 1;
     donations = amount;
     donorAmounts[author] = amount;
+    allDonors.push(author);
     highestDonors[0] = HighestDonor({ addr: author, lastDonatedAt: now });
   }
+  
+  function getDonorAmountByKey (uint key) public constant returns(donation uint) {
+    return donorAmounts[allDonors[key]];
+  }
 
   function receiveApproval(address from, uint256 value, address _tokenContract, bytes extraData) beforeDeadline {
-    require(address(token) == _tokenContract);
+    require(address(token) == _tokenContract); // necessary?
```

Comment:
No, should be enough just check what `msg.sender` is equal to address of token contract.